day1/adventcode2015_day1_refactor.py

```diff
@@ -0,0 +1,37 @@
+with open('input.txt', 'r') as floors:
+    visited_floors: str = floors.readline().strip()
+
+
+def count_by_character(input_character: str):
+
+    counter = 0
+    match input_character:
+        case '(':
+            counter += 1
+        case ')':
+            counter -= 1
+    return counter
+
+
+def count_santa_floor_and_basement_index(floors_input: str, target_floor: int = -1) -> tuple[int, int]:
+
+    brackets_counter: int = 0
+    basement_index: None = None
+
+    for floor_index, floor in enumerate(floors_input):
+
+        brackets_counter += count_by_character(floor)
+
+        if brackets_counter == target_floor and basement_index is None:
+            basement_index: int = floor_index + 1
```

Comment:
Added default argument as -1, can be changed. But maybe it will be better to pass it to function every time.